hyperspy/_signals/complex_signal.py

```diff
@@ -260,6 +261,59 @@ def angle(self, deg=False):
         return super().angle(angle, deg=deg)
     angle.__doc__ = ComplexSignal_mixin.angle.__doc__
 
+    def argand_diagram(self, size=[256, 256], display_range=None):
+        """
+        Calculate and plot Argand diagram of complex signal
+
+        Parameters
+        ----------
+        size : [int, int], optional
+            Size of the Argand plot in pixels
+            (Default: [256, 256])
+        display_range : array_like, shape(2,2) or shape(2,) optional
+            The position of the edges of the diagram
+            (if not specified explicitly in the bins parameters): [[xmin, xmax], [ymin, ymax]].
+            All values outside of this range will be considered outliers and not tallied in the histogram.
+            (Default: None)
+
+        Returns
+        -------
+        sap:
+            Argand diagram plot as Signal2D
+
+        Examples
+        --------
+        >>> import hyperspy.api as hs
+        >>> holo = hs.datasets.example_signals.object_hologram()
+        >>> ref = hs.datasets.example_signals.reference_hologram()
+        >>> w = holo.reconstruct_phase(ref)
+        >>> w.argand_diagram(display_range=[-3, 3]).plot()
+
+        """
+        im = self.imag.data.ravel()
+        re = self.real.data.ravel()
+
+        if display_range:
+            if np.asarray(display_range).shape == (2, ):
+                display_range = [[display_range[0], display_range[1]],
+                                 [display_range[0], display_range[1]]]
+            elif np.asarray(display_range).shape != (2, 2):
+                raise ValueError('display_range should be array_like, shape(2,2) or shape(2,).')
+
+        ap = np.histogram2d(re, im, bins=size, range=display_range)
+        sap = Signal2D(ap[0].T)
+        sap.metadata = self.metadata.deepcopy()
+        sap.metadata.General.title = 'Argand diagram of {}'.format(self.metadata.General.title)
+        sap.metadata.Signal.signal_type = 'signal2d'
```

Comment:
This should be blank, I think. I think signal_type is kindof deprecated since we have class names that cover the same use. @ericpre @francisco-dlp do you have thoughts on this?

Reply:
Yes, I don't think this should be used here.

Reply:
Just to clarify: `signal_type` is NOT deprecated but setting it is not necessary here. `signal_type` is actually quite important to define signal, see http://hyperspy.org/hyperspy-doc/current/dev_guide/writing_extensions.html#registering-a-new-basesignal-subclass.